### PR TITLE
GEODE-5971: Refactor CreateRegionCommand to extend SingleGfshCommand

### PR DESCRIPTION
**Pull request by aditya87**
- Store config via generating a RegionConfig object rather than an XML entity
- For clarity, rename getters that return a "Boolean" object rather than a boolean primitive as prefixed by "get" rather than "is". This communicates nullability to the caller. A bit of history here -- this code was originally generated by lombok (https://projectlombok.org/), which we subsequently backed out. But we like the clarity of this pattern.
- Create a "RegionConfigFactory" to generate a RegionConfig object from RegionFunctionArgs.
- Reduce null-related side effects in some setters in RegionFunctionArgs.

Signed-off-by: Peter Tran <ptran@pivotal.io>
Signed-off-by: Aditya Anchuri <aanchuri@pivotal.io>
Signed-off-by: Peter Tran <ptran@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [X] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [X] Is your initial contribution a single, squashed commit?

- [X] Does `gradlew build` run cleanly?

- [X] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/build.gradle [hunk 1]
```diff
@@ -15,10 +15,13 @@
  * limitations under the License.
  */
 
-
 apply plugin: 'antlr'
 apply plugin: 'me.champeau.gradle.jmh'
 
+repositories {
```
Comment (rhoughton-pivot) [inline geode-core/build.gradle]: Why was this needed? `mavenCentral` is already a repository for `allprojects`, added in the project root `build.gradle`

Comment (aditya87) [reply geode-core/build.gradle]: Sorry -- this might have been added when I temporarily brought in lombok, I realize it's unnecessary. Will remove.

**Comment by aditya87** [comment]
Withdrawing, will recreate once we fix some issues
